src/pamiq_core/trainer/base.py

```diff
@@ -0,0 +1,122 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+from pamiq_core.data import DataUser, DataUsersDict
+from pamiq_core.model import (
+    InferenceModelsDict,
+    TrainingModel,
+    TrainingModelsDict,
+)
+
+
+class Trainer(ABC):
```

Comment:
@zassou65535 状態も保存する機能を追加してほしい。
```suggestion
from pamiq_core.state_persistence import PersistentStateMixin

class Trainer(ABC, PersistentStateMixin):
```

テストもこうする

```py
class TestTrainer:
    def test_trainer_subclasses(self):
        assert issubclass(Trainer, PersistentStateMixin)
```

Reply:
https://github.com/MLShukai/pamiq-core/pull/63/commits/9ca4975fa3cac63d4e80a7371ec0626c1b0b3341 で修正